proparse/src/main/java/org/prorefactor/proparse/antlr4/ProgressLexer.java

```diff
@@ -108,39 +110,114 @@
   private int laChar;
   private int safetyNet = 0;
   private int sequence = 0;
-  private int sourceCounter;
+  private int sourceCounter = -1;
+
+  // From ProgressLexer
+  private Lexer lexer;
+  private IntegerIndex<String> filenameList = new IntegerIndex<>();
+  private final RefactorSession session;
+  private TokenSource wrapper;
+  private final String fileName;
 
   /**
    * An existing reference to the input stream is required for construction. The caller is responsible for closing that
    * stream once parsing is complete.
    */
-  public Preprocessor(String fileName, BufferedReader inStream, DoParse doParse) {
-    this.doParse = doParse;
-    this.ppSettings = doParse.getRefactorSession().getProparseSettings();
+  public ProgressLexer(RefactorSession session, String fileName) throws IOException {
+    LOGGER.trace("New ProgressLexer instance {}", fileName);
+    this.ppSettings = session.getProparseSettings();
+    this.session = session;
+    this.fileName = fileName;
 
     // Create input source with flag isPrimaryInput=true
-    sourceCounter = -1;
-    currFile = doParse.addFilename(fileName);
-    currentInput = new InputSource(++sourceCounter, inStream, true);
-    currentInput.fileIndex = currFile;
+    currFile = addFilename(fileName);
+    currentInput = new InputSource(++sourceCounter, new File(fileName), session.getCharset(), currFile, true);
     currentInclude = new IncludeFile(fileName, currentInput);
     includeVector.add(currentInclude);
     currSourceNum = currentInput.getSourceNum();
     lstListener = new ListingParser();
+    
+    lexer = new Lexer(this);
+    PostLexer postlexer = new PostLexer(lexer);
+    TokenSource filter1 = new TokenList(postlexer);
+    wrapper = new MultiChannelTokenSource(filter1);
   }
 
+  public String getFileName() {
+    return fileName;
+  }
+
+  public IntegerIndex<String> getFilenameList() {
+    return filenameList;
+  }
+
+  public String getFilename(int fileIndex) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') Rename method "getFilename" to prevent any misunderstanding/clash with method "getFileName" defined on line 146. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1845)